jl4-core/src/L4/Print.hs

```diff
@@ -171,33 +173,36 @@ instance LayoutPrinter a => LayoutPrinter (Import a) where
   printWithLayout = \case
     MkImport _ n -> "IMPORT" <+> printWithLayout n
 
-instance LayoutPrinter a => LayoutPrinter (Section a) where
+instance LayoutPrinter a => LayoutPrinter (Int, Section a) where
   printWithLayout = \case
-    MkSection _ _ Nothing _ ds    ->
-      vcat (fmap printWithLayout ds)
-    MkSection _ ps name maka ds ->
+    (i, MkSection _ Nothing _ ds)    ->
+      vcat (map (printWithLayout . (i + 1 ,)) ds)
+    (i, MkSection _ name maka ds) ->
       vcat $
-        [ pretty (replicate ps '§') <+>
+        [ pretty (replicate i '§') <+>
           case maka of
             Nothing  -> printWithLayout name
             Just aka -> printWithLayout name <+> printWithLayout aka
         ]
         <> case ds of
           [] -> mempty
-          _ -> fmap printWithLayout ds
+          _ -> map (printWithLayout . (i + 1 ,)) ds
 
 instance LayoutPrinter a => LayoutPrinter (Module  a) where
   printWithLayout = \case
-    MkModule _ _ sects ->
-      vcat (fmap printWithLayout sects)
+    MkModule _ _ sect -> printWithLayout (0 :: Int, sect)
 
 instance LayoutPrinter a => LayoutPrinter (TopDecl a) where
+  printWithLayout t = printWithLayout (1 :: Int, t)
+
+instance LayoutPrinter a => LayoutPrinter (Int, TopDecl a) where
```

Comment:
Let's give a name to this pair type. Something like `SectionedTopDecl`.

Reply:
Makes sense, I'll do that